File: python/generate.py
#!/usr/bin/python3

import argparse
import sys
from langgen import generate


def main(args):
    parser = argparse.ArgumentParser()
    parser.add_argument("fname", help="text file for corpus")
    parser.add_argument("-n", "--number", type=int, default=1,
                        help="number of random sequences")
    args = parser.parse_args(args)

    print(generate(args.fname, args.number))


if __name__ == '__main__':
    main(sys.argv[1:])

File: src/bin/generate.rs
extern crate rustc_serialize;
extern crate docopt;
extern crate langgen;

use std::path::Path;
use docopt::Docopt;

const USAGE: &'static str = "
Usage: generate [options] <file>...

Options:
    -n, --number=<x>  print x random sentences [default = 1]
    -s, --stats  print language courpus stats
    -h, --help  display this help and exit
    -v, --version  output version information and exit
";

#[derive(Debug, RustcDecodable)]
struct Args {
    arg_file: Vec<String>,
    flag_number: Option<usize>,
    flag_stats: bool
}

fn main() {
    let args: Args = Docopt::new(USAGE)
                            .and_then(|d| d.decode())
                            .unwrap_or_else(|e| e.exit());
    
    let num = args.flag_number.unwrap_or(1);
    let fnames = args.arg_file;
    let mut trigrams = langgen::Trigrams::new();
    for fname in fnames {
        let i=langgen::FileTokenizer::from_path(Path::new(&fname)).expect("Cannot open file").into_iter();
        trigrams.fill(i);
    }
    if args.flag_stats {
        trigrams.print_stats();
        return
    }
    for _i in 0..num {
        println!("{}", trigrams.random_sentence(1000));
    }
    
}